### PR TITLE
Bug fixes and improvements to static file serving

### PR DESCRIPTION
**Pull request by brianhelba**
* Allow static files to be served within Girder "Resource" classes
* Allow the content type of a static file to be explicity provided

### CODE REVIEW

File: girder/api/rest.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 ###############################################################################
 
 import cherrypy
+import cherrypy.lib
```
Comment (zachmullen) [inline girder/api/rest.py]: Since we already have `cherrypy` top level module imported, do we need to explicitly import `cherrypy.lib`?

Comment (brianhelba) [inline girder/api/rest.py]: Yes, we do need the explicit import. By design, CherryPy's root __init__.py file [1] does not do an "import lib" anywhere, so it's not available in the top "cherrypy" namespace.

Also, note that CherryPy's careful use of modules and imports presents a very clean public namespace. This is something Girder should eventually start to consider.

[1] https://github.com/cherrypy/cherrypy/blob/3.3.0/cherrypy/__init__.py

Comment (brianhelba) [inline girder/api/rest.py]: Oops, I'm totally wrong. I was using it with a different import style during my development, and that wasn't working. The extra explicit import of "cherrypy.lib" is not necessary.

I'll fix this.

**Comment by brianhelba** [comment]
Pushed an updated branch in response to @zachmullen feedback.

**Comment by zachmullen** [comment]
LGTM

**Pull request merged**